File: .tmp_py4013/batch_08_dump/2890_reshape_data_melt.js
// LeetCode 2890 - Reshape Data: Melt
// https://leetcode.com/problems/reshape-data-melt/

/**
 * @param {object[]} report
 * @return {object[]}
 */
var meltTable = function(report) {
    const out = [];
    for (const r of report) {
        if (Array.isArray(r)) {
            const product = r[0];
            for (let q = 1; q <= 4; q++) out.push({ product, quarter: 'quarter_' + q, sales: r[q] });
        } else {
            for (const q of ['quarter_1', 'quarter_2', 'quarter_3', 'quarter_4']) {
                out.push({ product: r.product, quarter: q, sales: r[q] });
            }
        }
    }
    return out;
};

File: .tmp_rs4013/rust_check_12/2890_reshape_data_melt.rs
struct Solution;
// LeetCode 2890 - Reshape Data: Melt
// https://leetcode.com/problems/reshape-data-melt/

impl Solution {
    pub fn melt_table(report: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let mut out = Vec::new();
        for r in report {
            if r.is_empty() {
                continue;
            }
            let product = r[0];
            for (q, &sales) in r.iter().enumerate().skip(1) {
                out.push(vec![product, q as i32, sales]);
            }
        }
        out
    }
}

fn main() {}
